docs/src/lib/indexing.md

```diff
@@ -238,12 +238,27 @@ The elements of a `GroupedDataFrame` are [`SubDataFrame`](@ref)s of its parent.
 This table presents return value types of calling `names`, `propertynames` and `keys`
 on types exposed to the user by DataFrames.jl:
 
-| Type                | `names`          | `propertynames`  | `keys`           |
-|---------------------|------------------|------------------|------------------|
-| `AbstractDataFrame` | `Vector{String}` | `Vector{Symbol}` | undefined        |
-| `DataFrameRow`      | `Vector{String}` | `Vector{Symbol}` | `Vector{Symbol}` |
-| `DataFrameRows`     | `Vector{String}` | `Vector{Symbol}` | vector of `Int`  |
-| `DataFrameColumns`  | `Vector{String}` | `Vector{Symbol}` | `Vector{Symbol}` |
-| `GroupedDataFrame`  | `Vector{String}` | tuple of fields  | `GroupKeys`      |
-| `GroupKeys`         | undefined        | tuple of fields  | vector of `Int`  |
-| `GroupKey`          | `Vector{String}` | `Vector{Symbol}` | `Vector{Symbol}` |
+| Type                | `names`          | `propertynames`  | `keys`           | `length`  | `ndims` |
+|---------------------|------------------|------------------|------------------|-----------|---------|
+| `AbstractDataFrame` | `Vector{String}` | `Vector{Symbol}` | undefined        | undefined | `2`     |
+| `DataFrameRow`      | `Vector{String}` | `Vector{Symbol}` | `Vector{Symbol}` | `Int`     | `1`     |
+| `DataFrameRows`     | `Vector{String}` | `Vector{Symbol}` | vector of `Int`  | `Int`     | `1`     |
+| `DataFrameColumns`  | `Vector{String}` | `Vector{Symbol}` | `Vector{Symbol}` | `Int`     | `1`     |
+| `GroupedDataFrame`  | `Vector{String}` | tuple of fields  | `GroupKeys`      | `Int`     | `1`     |
+| `GroupKeys`         | undefined        | tuple of fields  | vector of `Int`  | `Int`     | `1`     |
+| `GroupKey`          | `Vector{String}` | `Vector{Symbol}` | `Vector{Symbol}` | `Int`     | `1`     |
+
+Additionally the above types `T` (i.e. `AbstractDataFrame`, `DataFrameRow`, `DataFrameRows`,
+`DataFrameColumns`, `GroupedDataFrame`, `GroupKeys`, `GroupKey`) the following methods are defined:
+* `size(::T)` returning a `Tuple` of `Int`.
+* `size(::T, ::Integer)` returning an `Int`.
+* `axes(::T)` returning a `Tuple` of `Int` vectors.
+* `axes(::T, ::Integer)` returning an `Int` vector.
```

Comment:
When the dimension is invalid an error is thrown, at least for `AbstractDataFrame`, right?

Reply:
Yes, it is thrown except for `DataFrameRows` and `GroupKeys` which are `AbstractVector`s so:
```
julia> axes(df, 3)
ERROR: ArgumentError: DataFrames only have two dimensions

julia> axes(eachrow(df), 3)
Base.OneTo(1)

julia> axes(keys(groupby(df, names(df, 1))), 3)
Base.OneTo(1)
```

I find this a bit inconsistent in Base, but this is how it works.

Reply:
Sorry, my point was that this should probably be mentioned like for `firstindex`.

Reply:
added